extensions/AdvancedSearch/populateCategorySearch.inc

```diff
@@ -7,7 +7,7 @@
 
 define( 'REPORTING_INTERVAL', 1000 );
 
-function populateCategorySearch( $begin, $maxlag, $throttle, $force ) {
+function populateCategorySearch( $begin, $throttle, $force ) {
```

Comment:
The changes to fix the specific issue for this ticket are in 2 files: GlobalFunctions.php and Loadbalancer.php. The rest of this is me cleaning up the unused and deprecated argument $maxLag to wfWaitForSlaves.